### PR TITLE
Add bitwise and operator.

### PR DESCRIPTION
**Pull request by ummarikar**
Add bitwise and operator - add tests for Ints and BigInts and tests for when the argument is not an Int.

### CODE REVIEW

File: src/lib/vm/objects/mod.rs [hunk 1]
```diff
@@ -90,6 +90,11 @@ pub trait Obj: std::fmt::Debug + abgc::GcLayout {
         unimplemented!();
     }
 
+    /// Produce a new `Val` which performs a bitwise and with `other` and this
```
Comment (ltratt) [inline src/lib/vm/objects/mod.rs]: Missing "." at end of sentence.

**Comment by ummarikar** [comment]
Done (apologies for missing that!).

**Comment by ltratt** [comment]
Quick question: have you checked this against another SOM (e.g. Java SOM)?

**Comment by ummarikar** [comment]
> Quick question: have you checked this against another SOM (e.g. Java SOM)?

Yes indeed. All features I implement I check them against Java SOM before committing the changes. I run against Java SOM, copy the output into my tests, then implement the feature to get the tests to pass. Is that the right way to go about it?

**Comment by ltratt** [comment]
That's the right way to do it IMHO (though you should keep a questioning mind -- Java SOM isn't always correct itself!).

Please squash.

**Comment by ummarikar** [comment]
Understood, I will definitely keep this in mind.

**Comment by ummarikar** [comment]
Squashed.

**Pull request merged**